js/src/tags.js

```diff
@@ -0,0 +1,467 @@
+        self.button_class_edit = 'js-listable-item-admin-edit-button';
+        self.button_class_reset = 'js-listable-item-admin-reset-button';
+        self.button_class_delete = 'js-listable-item-admin-delete-button';
+        self.button_next = $('#js-listable-item-admin-nav-next');
+        self.button_previous = $('#js-listable-item-admin-nav-previous');
+        self.button_save = $('#js-listable-item-admin-save');
+        self.button_cancel = $('#js-listable-item-admin-cancel');
+        self.limit = 20;
+        self.offset = 0;
+        self.page = 1;
+        self.page_count = 1;
+
+        self.html_template = $('#js-listable-item-admin-item-template');
+        self.template_heading = $('#js-lia-template-heading');
+
+        self.localised_edit = $('#js-lia-loc-edit').text();
+        self.localised_delete = $('#js-lia-loc-delete').text();
+        self.localised_save_edit = $('#js-lia-loc-save-edit').text();
+        self.localised_reset = $('#js-lia-loc-reset').text();
+
+        self.items_list = $('#js-listable-item-admin-list');
+        //TODO merge the items and store statuses as properties
+        self.loaded_items = {};
+        self.unlocked_items = {};
+        self.modified_items = {};
+        self.item_errors = {};
+        self.displayed_ids = [];
+        self.deleted_items_ids = [];
+        self.new_item_ids = [];
+        self.editable_field_names = [];
+        self.new_offset = 1;
+        self.saveUrl = '/api/thankyou/v2/tags';
+        self.errors_banner = $('#js-listable-item-admin-errors');
+        self.no_results_banner = $('#js-listable-item-admin-no-results');
+
+        self.items_list.on('click', '.' + self.button_class_edit, function () {
+            self.toggleEditMode(this.closest('.' + self.row_class).getAttribute('data-id'));
+        });
+
+        self.items_list.on('click', '.' + self.button_class_reset, function () {
+            var id = this.closest('.' + self.row_class).getAttribute('data-id');
+            self.resetEdit(id);
+            self.refreshDisplay();
+        });
+
+        self.items_list.on('click', '.' + self.button_class_delete, function () {
+            var id = this.closest('.' + self.row_class).getAttribute('data-id');
+            self.deleteItem(id);
+        });
+
+        self.button_create.click(function () {
+            self.createNew();
+        });
+
+        self.button_next.click(function () {
+            self.changePage(self.page + 1)
+        });
+
+        self.button_previous.click(function () {
+            self.changePage(self.page - 1)
+        });
+
+        self.button_save.click(function () {
+            self.save();
+        });
+
+        self.button_cancel.click(function () {
+            self.resetAll();
+        });
+
+        self.loadPageCount();
+    };
+
+    ListableItemsAdmin.prototype.createNew = function () {
+        var template = $($.parseHTML(_.template(this.html_template.html())({})));
```

Comment:
Another template that could be precompiled and benefit from placeholders

Reply:
Fixed in ef1176accf5c411a39af7103186807ffe773720a